src/Classes/TradeQueryGenerator.lua

```diff
@@ -0,0 +1,698 @@
+                    goto nextModLine
+                end
+
+                self.modData[modType][statOrder] = { tradeMod = tradeMod, specialCaseData = specialCaseData }
+            end
+
+            -- tokenize the numerical variables for this mod and store the sign if there is one
+            local tokens = { }
+            local poundPos, tokenizeOffset = 0, 0
+            while true do
+                poundPos = self.modData[modType][statOrder].tradeMod.text:find("#", poundPos + 1)
+                if poundPos == nil then
+                    break
+                end
+                startPos, endPos, sign, min, max = modLine:find("([%+%-]?)%(?(%d+%.?%d*)%-?(%d*%.?%d*)%)?", poundPos + tokenizeOffset)
+
+                if endPos == nil then
+                    logToFile("Error extracting tokens from '%s' for tradeMod '%s'", modLine, self.modData[modType][statOrder].tradeMod.text)
+                    goto nextModLine
+                end
+
+                tokenizeOffset = tokenizeOffset + (endPos - startPos)
+                table.insert(tokens, min)
+                table.insert(tokens, #max > 0 and tonumber(max) or tonumber(min))
+                if sign ~= nil then
+                    self.modData[modType][statOrder].sign = sign
+                end
+            end
+
+            if #tokens ~= 0 and #tokens ~= 2 and #tokens ~= 4 then
+                logToFile("Unexpected # of tokens found for mod: %s", mod[i])
+                goto nextModLine
+            end
+
+            -- Update the min and max values available for each item category
+            for category, tags in pairs(itemCategoryTags) do
+                if canModSpawnForItemCategory(mod, tags) then
+                    if self.modData[modType][statOrder][category] == nil then
+                        self.modData[modType][statOrder][category] = { min = 999999, max = -999999 }
+                    end
+
+                    local modRange = self.modData[modType][statOrder][category]
+                    if #tokens == 0 then
+                        modRange.min = 1
+                        modRange.max = 1
+                    elseif #tokens == 2 then
+                        modRange.min = math.min(modRange.min, tokens[1])
+                        modRange.max = math.max(modRange.max, tokens[2])
+                    elseif #tokens == 4 then
+                        modRange.min = math.min(modRange.min, (tokens[1] + tokens[3]) / 2)
+                        modRange.max = math.max(modRange.max, (tokens[2] + tokens[4]) / 2)
+                    end
+                end
+            end
+            ::nextModLine::
+        end
+        ::continue::
+    end
+end
+
+function TradeQueryGeneratorClass:InitMods()
+    local queryModFilePath = "Data/QueryMods.lua"
+
+    local file = io.open(queryModFilePath,"r")
+    if file then
+        file:close()
+        self.modData = LoadModule(queryModFilePath)
+        return
+    end
```

Comment:
All you have to do is delete the file... and it will auto-regenerate it. Only reason we save it is it takes a good 10-20 seconds on a decent computer and people complain when things "are slow".

Reply:
Yeah for sure, I like saving the file off for speed.  I'm guessing these mods will get added to each time one appears on the trade site, so we can't just push it out with an update.  Maybe it's possible to store a hash of the json, and if the hash is different than the file, regenerate it (or simply add to it?  Not sure which would be better).  The only downside is requesting stats is one less request we have for actual queries

Reply:
We should only do this approach on the PoB Development side (not as part of client distributable). To do a hash of a file you have to pull it all down. It's a hefty file.